### PR TITLE
PMM-10072 Filter out profiler collection

### PR DESCRIPTION
**Pull request by ShashankSinha252**
[PMM-10072](https://jira.percona.com/browse/PMM-10072) Filter out profiler collection for monitoring

### CODE REVIEW

**Review by JiriCtvrtka — APPROVED** [review]

File: exporter/common.go [hunk 1]
```diff
@@ -63,7 +66,15 @@ func listCollections(ctx context.Context, client *mongo.Client, database string,
 		return nil, errors.Wrap(err, "cannot get the list of collections for discovery")
 	}
 
-	return collections, nil
+	filteredCollections := []string{}
+	for _, collection := range collections {
+		if collection == systemCollections[0] {
```
Comment (dliakhov) [inline exporter/common.go:71]: if there is only one element in the systemCollections slice, could we use it as a constant variable?

Or if you plan to add more elements to `systemCollections` slice later could you please write this code which will work if the slice has two or more elements, but not only one?

File: exporter/common.go [hunk 2]
```diff
@@ -29,7 +29,10 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-var systemDBs = []string{"admin", "config", "local"} //nolint:gochecknoglobals
+var (
+	systemDBs         = []string{"admin", "config", "local"} //nolint:gochecknoglobals
+	systemCollections = []string{"system.profile"}
+)
 
 func listCollections(ctx context.Context, client *mongo.Client, database string, filterInNamespaces []string) ([]string, error) {
```
Comment (dliakhov) [inline exporter/common.go:37]: is it possible to add tests for the new behavior of filtering collections?

File: exporter/common.go [hunk 3]
```diff
@@ -201,6 +202,21 @@ func listAllCollections(ctx context.Context, client *mongo.Client, filterInNames
...
+	}
+
+	filtered := make([]string, 0, len(cols))
+	for _, col := range cols {
+		if systemCollections[col] {
+			continue
+		}
+		filtered = append(filtered, col)
+	}
+	return filtered
```
[the remark this answers is not in the document]
Comment (ShashankSinha252) [reply exporter/common.go:219]: Resolved

File: exporter/common.go [hunk 4]
```diff
@@ -201,6 +203,22 @@ func listAllCollections(ctx context.Context, client *mongo.Client, filterInNames
 	return namespaces, nil
 }
 
+func filterSystemCollections(cols []string) []string {
+	systemCollections := map[string]bool{
```
Comment (dliakhov) [inline exporter/common.go:207]: could you please move out initialization of this map? I'm not sure that it makes sense to create every time this map when `filterSystemCollections` is called.

**Comment by BupycHuk** [comment]
closing it as it was done in #917

**Pull request closed without merging**